Analytiikka paallystyskohteiden haku parannukset

jpluomajoki-solita opened the pull request.

jpluomajoki-solita committed:
Vaihtoehtoinen yha-skeema, jossa pot2-tiedot.

jpluomajoki-solita committed:
Massan runko- ja lisäaine pakollisiksi

jpluomajoki-solita committed:
Lisää ajorata, kaista ja pituus päällystyskohteisiin

jpluomajoki-solita committed:
Lisää pot2-tiedot analytiikan skeemoihin, poista desimaalien multipleOf-validoinnit (eivät toimi kunnolla)

jpluomajoki-solita committed:
Poista tarpeettomia tietoja, lisää yha-lähetystietoja

jpluomajoki-solita committed:
päivitä esimerkki ja pakollisuuksia skeemassa

jpluomajoki-solita committed:
Merge branch 'develop' into analytiikka-paallystyskohteiden-haku-parannukset

jpluomajoki-solita committed:
Lisää paikkausten hakurajapinnan skeema

jpluomajoki-solita committed:
Lisää hoitourakoiden paikkausten hakurajapinta

jpluomajoki-solita committed:
Lisää päällystysurakoiden hakurajapinta analytiikalle

jpluomajoki-solita committed:
Lisää luotu-kenttä ylläpitokohdeosatauluihin

jpluomajoki-solita committed:
Päällystyskohteiden hakurajapinta

jpluomajoki-solita committed:
Lisää aikataulujen rajapinta

jpluomajoki-solita committed:
Tee päällystysilmoitusten rajapinta

jpluomajoki-solita committed:
Poista kommentti

jpluomajoki-solita committed:
Lisää hoidon paikkausten rajapinta

jpluomajoki-solita committed:
Päivitä esimerkit

jpluomajoki-solita committed:
Nosta migraatio

jpluomajoki-solita committed:
Merge branch 'develop' into analytiikka-paallystyskohteiden-haku-parannukset

solita-staskila commented on src/clj/harja/kyselyt/paallystys.sql:
```diff
@@ -834,3 +834,241 @@ UPDATE pot2_mk_urakan_murske
+       pk_osa.tr_loppuetaisyys     AS "tr-loppuetaisyys",
+       pk_osa.tr_kaista            AS "tr-kaista",
+       pk_osa.tr_ajorata           AS "tr-ajorata",
+       -- pituus lasketaan clj-puolella.
```
Meillä on olemassa myös tr osoitteen pituuden laskenta tietokantaproseduuri mikä on todettu monta kertaa nopeammaksi kun lasku clj- puolella. 
Se että onko sillä tässä kauheasti väliä, voi itse miettiä, mutta kommenttina vaan että tällainen on olemassa. Jos tässä kyselyssä on mahdollista että kohteita tulee olemaan paljon, ehdottaisin että tehdään pituuden laskenta tietokannan puolella 

```sql
(SELECT * FROM laske_tr_osoitteen_pituus(
    ypk.tr_numero, 
    ypk.tr_alkuosa, 
    ypk.tr_alkuetaisyys, 
    ypk.tr_loppuosa,
    ypk.tr_loppuetaisyys)) AS "pituus",
```

jpluomajoki-solita replied:
Kuulostaa oikein hyvältä, kiitos tästä!

jpluomajoki-solita committed:
Lisää urakan haun skeema

jpluomajoki-solita committed:
Laske pituus SQL:ssä, korjaa muutamia avaimia

solita-staskila commented on src/clj/harja/palvelin/integraatiot/api/analytiikka.clj:
```diff
@@ -686,6 +862,51 @@
+        (kasittele-get-kutsu db integraatioloki :analytiikka-hae-hoidon-paikkaukset parametrit
+          json-skeemat/+analytiikka-hoidon-paikkaukset-haku-vastaus+
+          (fn [parametrit _kayttaja db]
+            (hae-hoidon-paikkaukset db parametrit))
```
Onko näissä reittien julkaisussa ollut tarkoitus _kayttaja parametri laittaa vaan alaviivaksi? Ei ainakaan funktioon tarvitse passata

jpluomajoki-solita replied:
Kyllä, alaviivalla merkitään parametri tai muuttuja, jota ei käytetä. Monesti käytetään vaan alaviivaa, mutta sen voi myös nimetä tällä tavalla.

solita-staskila reviewed (approved):
Ainakin koodin puolesta näyttää hyvälle, niin hyvälle miltä clojure voi näyttää 🍝

jpluomajoki-solita committed:
Merge branch 'develop' into analytiikka-paallystyskohteiden-haku-parannukset

jpluomajoki-solita committed:
Korjaa testi

 - hajonnut, koska odotettiin tiettyä järjestystä ilman order by-lausetta, ja migraatiossa päivitetään kantaa.
